pss/pss.go

```diff
@@ -144,681 +120,32 @@ type Pss struct {
-	defer metrics.GetOrRegisterResettingTimer("pss/execute-handlers", nil).UpdateSince(time.Now())
-
-	handlers := p.getHandlers(topic)
-	peer := p2p.NewPeer(enode.ID{}, hex.EncodeToString(from), []p2p.Cap{})
-	for _, h := range handlers {
-		if !h.caps.raw && raw {
-			log.Warn("norawhandler")
-			continue
-		}
-		if !h.caps.prox && prox {
-			log.Warn("noproxhandler")
-			continue
-		}
-		err := (h.f)(payload, peer, asymmetric, keyid)
-		if err != nil {
-			log.Warn("Pss handler failed", "err", err)
-		}
-	}
-}
-
-// will return false if using partial address
-func (p *Pss) isSelfRecipient(msg *message.Message) bool {
-	return bytes.Equal(msg.To, p.Kademlia.BaseAddr())
-}
-
-// test match of leftmost bytes in given message to node's Kademlia address
-func (p *Pss) isSelfPossibleRecipient(msg *message.Message, prox bool) bool {
-	local := p.Kademlia.BaseAddr()
-
-	// if a partial address matches we are possible recipient regardless of prox
-	// if not and prox is not set, we are surely not
-	if bytes.Equal(msg.To, local[:len(msg.To)]) {
-
-		return true
-	} else if !prox {
-		return false
-	}
-
-	depth := p.NeighbourhoodDepth()
-	po, _ := network.Pof(p.Kademlia.BaseAddr(), msg.To, 0)
-	log.Trace("selfpossible", "po", po, "depth", depth)
-
-	return depth <= po
-}
-
-/////////////////////////////////////////////////////////////////////
-// SECTION: Message sending
-/////////////////////////////////////////////////////////////////////
-
-func (p *Pss) enqueue(msg *message.Message) {
-	defer metrics.GetOrRegisterResettingTimer("pss/enqueue", nil).UpdateSince(time.Now())
-
-	// TODO: create and enqueue in one outbox method
-	outboxMsg := p.outbox.NewOutboxMessage(msg)
-	p.outbox.Enqueue(outboxMsg)
-}
-
-// Send a raw message (any encryption is responsibility of calling client)
-//
-// Will fail if raw messages are disallowed
-func (p *Pss) SendRaw(address PssAddress, topic message.Topic, msg []byte, messageTTL time.Duration) error {
-	defer metrics.GetOrRegisterResettingTimer("pss/send/raw", nil).UpdateSince(time.Now())
-
-	if err := validateAddress(address); err != nil {
-		return err
-	}
-
-	pssMsgParams := message.Flags{
-		Raw: true,
-	}
-
-	pssMsg := message.New(pssMsgParams)
-	pssMsg.To = address
-	pssMsg.Expire = uint32(time.Now().Add(messageTTL).Unix())
-	pssMsg.Payload = msg
-	pssMsg.Topic = topic
-
-	p.addFwdCache(pssMsg)
-
-	p.enqueue(pssMsg)
-	return nil
-}
-
-// Send a message using symmetric encryption
-//
-// Fails if the key id does not match any of the stored symmetric keys
-func (p *Pss) SendSym(symkeyid string, topic message.Topic, msg []byte) error {
-	symkey, err := p.GetSymmetricKey(symkeyid)
+// Send a message without encryption
+// Generate a trojan chunk envelope and is stored in localstore for desired targets to mine this chunk and retrieve message
+func Send(ctx context.Context, localStore chunk.Store, targets [][]byte, topic string, msg []byte) (chunk.Chunk, error) {
```

Comment:
the `topic` received here should be of the `trojan.Topic` type, not a `string`

Reply:
I agree the correct way is to construct the topic outside before calling Sent.
Thanks for pointing it out.

Fixed